Sources/SwiftSyntax/SourceLocation.swift

```diff
@@ -10,26 +10,75 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// Represents a source location in a Swift file.
-public struct SourceLocation: Codable {
+/// Represent the user-facing part of SourceLocation that can be calculated
+/// on demand.
+struct ComputedLocation: Codable {
   /// The line in the file where this location resides. 1-based.
-  public let line: Int
+  let line: Int
 
   /// The UTF-8 byte offset from the beginning of the line where this location
   /// resides. 1-based.
-  public let column: Int
+  let column: Int
+
+  /// The file in which this location resides.
+  let file: String
```

Comment:
it's not clear to me why `file` is part of `ComputedLocation`, and not of `SourceLocation`.

Reply:
I think file name is similar to line and column in terms of being user facing (we need it only when we print the location to command line). Also I plan to unify Sourcelocation with absolute path, which doesn’t need file name at all.